Assets/Script/Gameplay/HUD/Practice/PracticeSectionView.cs

```diff
@@ -101,5 +105,157 @@ public void UpdateView()
                 _background.color = Color.clear;
             }
         }
+
+        private string ParseSectionName(string sectionName)
+        {
+            // Best-effort attempt to convert [section]-based practice sections to the expected format
+            sectionName = sectionName.ToLower().Replace(' ', '_').Replace("guitar", "gtr");
+
+            // Handle letter-based sections like "A section" (prc_a) and "B section 3" (prc_b3)
+            if (IsLetterBasedSectionName(sectionName))
+            {
+                var letterBasedName = Localize.KeyFormat(
+                    ("Gameplay", "Practice", "SectionFormats", "LetterSection"),
+                    char.ToUpper(sectionName[0])
+                );
+
+                if (sectionName.Length == 1)
+                {
+                    return letterBasedName;
+                }
+
+                return Localize.KeyFormat(
+                    ("Gameplay", "Practice", "SectionFormats", "WithNumber"),
+                    letterBasedName,
+                    sectionName[1..].TrimStart('_')
+                );
+            }
+
+            (var name, var number) = DeriveNameAndNumber(sectionName);
+
+            var key = Localize.MakeKey("Gameplay", "Practice", "Sections", name);
+            var localizedName = Localize.Key(key);
+
+            if (localizedName == key)
+            {
+                // No localization for the section name, so return unlocalized
+                return sectionName.Replace('_', ' ');
+            }
+
+            if (number != null)
+            {
+                return Localize.KeyFormat(("Gameplay", "Practice", "SectionFormats", "WithNumber"), localizedName, number);
+            }
+
+            return localizedName;
+        }
+
+        /** 
+         * Whether the string follows the convention for letter-based section names
+         */
+        private bool IsLetterBasedSectionName(string text)
+        {
+            if (text.Length == 0 || !text[0].IsAsciiLetterLower())
+            {
+                return false;
+            }
+
+            if (text.Length == 1)
+            {
+                // e.g. "a" for "A section"
+                return true;
+            }
+
+            if (text.Length == 2)
+            {
+                // e.g. "a1" for "A section 1"
+                return text[1].IsAsciiDigit();
+            }
+
+            if (text[1] == '_')
+            {
+                // e.g. "a_b" for "A section B"
+                return text.Length == 3 && text[2].IsAsciiLetterLower();
+            }
+
+            if (text[1..^1].All(c => c.IsAsciiDigit()))
```

Comment:
Also using LINQ here. In this case the alternative would be more like:
```suggestion
            bool isDigit = true;
            for (int i = 1;i < text.Length - 1; i++)
            {
                if (!text[i].IsAsciiDigit()) 
                {
                    isDigit = false;
                    break;
                }
            }
            if (isDigit) {
```